### PR TITLE
Add support for `UserPool` resource

### PR DESCRIPTION
**Pull request by michaelhtm**
Issue [#1211](https://github.com/aws-controllers-k8s/community/issues/1211)

Description of changes:
* Generated `UserPool` resource
* Added e2e tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: apis/v1alpha1/generator.yaml [hunk 1]
```diff
@@ -4,7 +4,25 @@ ignore:
...
+            UserPoolId: Id
+        UpdateUserPool:
+          input_fields:
+            UserPoolId: Id
+        DeleteUserPool:
+          input_fields:
+            UserPoolId: Id
+        CreateUserPool:
+          input_fields:
+            PoolName: Name
```
Comment (a-hilaly) [inline apis/v1alpha1/generator.yaml:26]: Nice renames!

File: apis/v1alpha1/generator.yaml [hunk 2]
```diff
@@ -4,7 +4,25 @@ ignore:
...
+    tags:
+      ignore: true
```
Comment (a-hilaly) [inline apis/v1alpha1/generator.yaml:28]: Ideally we need tags for tracking purposes, wanna do this in a seperate PR?

**Review by a-hilaly — COMMENTED** [review]
Nice @michaelhtm ! 👍 

i have two minor comments below

File: apis/v1alpha1/generator.yaml (hunk 2 above)
Comment (michaelhtm) [reply apis/v1alpha1/generator.yaml:28]: i can add it on here...looks like it's a separate api for find and update...

Comment (michaelhtm) [reply apis/v1alpha1/generator.yaml]: added

**Review by a-hilaly — APPROVED** [review]
🚀 
/lgtm

**Pull request merged**